vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/dataview/CheckboxGroupListDataView.java

```diff
@@ -15,34 +15,44 @@
  */
 package com.vaadin.flow.component.checkbox.dataview;
 
+import java.util.List;
+
 import com.vaadin.flow.component.checkbox.CheckboxGroup;
 import com.vaadin.flow.data.provider.AbstractListDataView;
-import com.vaadin.flow.data.provider.DataController;
-
-import java.util.List;
+import com.vaadin.flow.data.provider.DataProvider;
+import com.vaadin.flow.function.SerializableSupplier;
 
 /**
- * {@link CheckboxGroup} component list data view implementation which takes into
+ * {@link CheckboxGroup} component list data view implementation which takes
+ * into
  * account the component-specific and common list data API.
  *
  * @param <T>
- *        data type
+ *         data type
  */
-public class CheckboxGroupListDataView<T> extends AbstractListDataView<T> implements CheckboxGroupDataView<T> {
+public class CheckboxGroupListDataView<T> extends AbstractListDataView<T>
+        implements CheckboxGroupDataView<T> {
+
+    SerializableSupplier<CheckboxGroup<T>> checkboxSupplier;
```

Comment:
It should be fine to just switch to the direct component reference

Reply:
done